File: frontend/src/types/job.ts
export type JobStatus = 'draft' | 'published' | 'closed' | 'archived';
export type EmploymentType = 'full_time' | 'part_time' | 'contract' | 'internship' | 'remote';

export interface Job {
  id: string; title: string; department?: string; location?: string;
  employment_type: EmploymentType; description: string;
  requirements: string[]; nice_to_have: string[]; keywords: string[];
  salary_min?: number; salary_max?: number; salary_currency: string;
  status: JobStatus; public_slug: string;
  total_applications: number; avg_score?: number;
  published_at?: string; closes_at?: string; created_at: string; updated_at: string;
}

export interface JobCreate {
  title: string; department?: string; location?: string;
  employment_type: EmploymentType; description: string;
  requirements: string[]; nice_to_have: string[]; keywords?: string[];
  salary_min?: number; salary_max?: number; salary_currency: string;
  closes_at?: string;
}

export type JobUpdate = Partial<JobCreate>;

export interface JobStats {
  total: number; received: number; screening: number;
  interview: number; offer: number; hired: number; rejected: number;
  avg_score: number;
}

export interface PublicJob extends Omit<Job, 'total_applications' | 'avg_score' | 'status'> {}

File: backend/app/schemas/job.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import List, Optional

class JobCreate(BaseModel):
    title: str
    department: Optional[str] = None
    location: Optional[str] = None
    employment_type: str = "full_time"
    description: str
    requirements: List[str] = []
    nice_to_have: List[str] = []
    keywords: List[str] = []
    salary_min: Optional[int] = None
    salary_max: Optional[int] = None
    salary_currency: str = "USD"

class JobUpdate(BaseModel):
    title: Optional[str] = None
    department: Optional[str] = None
    location: Optional[str] = None
    employment_type: Optional[str] = None
    description: Optional[str] = None
    requirements: Optional[List[str]] = None
    nice_to_have: Optional[List[str]] = None
    keywords: Optional[List[str]] = None
    salary_min: Optional[int] = None
    salary_max: Optional[int] = None
    salary_currency: Optional[str] = None

class JobOut(BaseModel):
    id: str
    title: str
    department: Optional[str]
    location: Optional[str]
    employment_type: str
    description: str
    requirements: List[str]
    nice_to_have: List[str]
    keywords: List[str]
    salary_min: Optional[int]
    salary_max: Optional[int]
    salary_currency: str
    status: str
    public_slug: Optional[str]
    closes_at: Optional[datetime]
    published_at: Optional[datetime]
    created_at: datetime
    
    # Extras from joins
    total_applications: Optional[int] = 0
    apps_by_stage: Optional[dict] = {}

    model_config = ConfigDict(from_attributes=True)

class PublicJobOut(BaseModel):
    id: str
    title: str
    department: Optional[str]
    location: Optional[str]
    employment_type: str
    description: str
    requirements: List[str]
    nice_to_have: List[str]
    salary_min: Optional[int]
    salary_max: Optional[int]
    salary_currency: str
    public_slug: str
    closes_at: Optional[datetime]
    published_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)

class PublicCompanyBoardOut(BaseModel):
    tenant: dict
    jobs: List[PublicJobOut]
